src/components/inline_edit/inline_edit_form.test.tsx

```diff
@@ -524,4 +524,105 @@ describe('EuiInlineEditForm', () => {
       });
     });
   });
+
+  describe('inline edit as a controlled component', () => {
+    const onSave = jest.fn();
+    const onChange = jest.fn();
+    const onCancel = jest.fn();
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    const controlledInlineEditFormProps: EuiInlineEditFormProps = {
+      ...requiredProps,
+      value: 'Hello World!',
+      onChange: onChange,
+      onCancel: onCancel,
+      inputAriaLabel: 'Edit inline',
+      sizes: MEDIUM_SIZE_FORM,
+      children: (readModeValue) => readModeValue,
+    };
+
+    it('renders the passed value in read mode', () => {
+      const { getByTestSubject, getByText } = render(
+        <EuiInlineEditForm {...controlledInlineEditFormProps} />
+      );
+
+      expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
+      expect(getByText('Hello World!')).toBeTruthy();
+    });
+
+    it('renders the passed value in edit mode', () => {
+      const { getByTestSubject } = render(
+        <EuiInlineEditForm
+          {...controlledInlineEditFormProps}
+          startWithEditOpen={true}
+        />
+      );
+
+      expect(getByTestSubject('euiInlineEditModeInput')).toHaveValue(
+        'Hello World!'
+      );
+    });
+
+    it('calls controlled onChange when edit mode value changes', () => {
+      const { getByTestSubject } = render(
+        <EuiInlineEditForm
+          {...controlledInlineEditFormProps}
+          startWithEditOpen={true}
+        />
+      );
+
+      const mockChangeEvent = { target: { value: 'changed' } };
+      fireEvent.change(
+        getByTestSubject('euiInlineEditModeInput'),
+        mockChangeEvent
+      );
+      waitFor(() => {
+        expect(onChange).toHaveBeenCalledWith(mockChangeEvent);
+      });
+    });
+
+    it('calls controlled onCancel when changes are cancelled', () => {
+      const { getByTestSubject } = render(
+        <EuiInlineEditForm
+          {...controlledInlineEditFormProps}
+          startWithEditOpen={true}
+          onSave={onSave}
+        />
+      );
+
+      fireEvent.click(getByTestSubject('euiInlineEditModeCancelButton'));
```

Comment:
We still need an extra `rerender` with an updated value to confirm the cancel callback is called with the old value:
```suggestion
      const { rerender, getByTestSubject } = render(
        <EuiInlineEditForm
          {...controlledInlineEditFormProps}
          startWithEditOpen={true}
          onSave={onSave}
        />
      );
      // Update controlled `value`
      rerender(
        <EuiInlineEditForm
          {...controlledInlineEditFormProps}
          startWithEditOpen={true}
          onSave={onSave}
          value="Updated value"
        />
      );

      fireEvent.click(getByTestSubject('euiInlineEditModeCancelButton'));
```